### DISCUSSION TITLE
Write to list in object without replacing other objects in list

### DISCUSSION THREAD

**Discussion by realrkace** [isar/isar#973] [Help]
I am using isar db as a middleware for my backend. I need the system it to perform computationally expensive tasks for which I chose this. It has been performing great so far. But I have a question:

My data is structured as follows:
```
{
    "id": 10001,
    outlet: [
        {
            "stock": 12,
            "price": 340
            "cost": 300,
            "storeId": 5
        },
        {
            "stock": 15,
            "price": 30
            "cost": 25,
            "storeId": 4
        }
    ]
}
```
Now from my api, I receive updates as the following:
```
{
    "id": 10001,
    outlet: [
        {
            "stock": 3, //the field that changed
            "price": 340
            "cost": 300,
            "storeId": 5
        },
    ]
}
```

As you can see only one field of the object in the list changed. But when I update ie. ```isar.iTems.putAll(newList)```, it deletes the other objects in the list, ie. object containing storeId: 4. How can this be solved? Are there any plans to include a **merge function** to do the same, basically compare existing data structure to incoming and update the values based on changed data in the incoming request instead of replacing it.

**Comment by simc** [comment]
Isar has no way of knowing what your intention is. Maybe the backend data is more up-to-date and the other objects in the list actually got deleted. You probably have to change your model to support this kind of synchronization.

**Reply by realrkace** [reply]
We're using the APIs to slowly migrate away from old infrastructure so there is a lot of friction. But in the documentation it is mentioned that the document will update if the id already exists, so I thought it'd behave that way. This functionality is critical for my use case so I thought it'd make sense to use isar. What change would you recommend in this model? Sorry I'm new to this, mongodb has a way to do this via aggregation pipelines, not sure how I should change it...

**Reply by jbrisko** [reply]
I think what @simc is saying is that Isar can't decide when you are using putAll() if you want your returned list from the API to be the entire dataset in the table, or if you just want to update the passed records into the table. You have to make those decisions. In our implementation for each endpoint coming back we decide whether it's a delete operation or a modify operation and interact with Isar accordingly. Either deleting old records, or going one-by-one through the result set using a put() statement on the single record, which will update the single record.

Hope this helps.

**Reply by realrkace** [reply]
@jbrisko I get it. I am reworking the structure. There are 2 issues, each object technically has 2 ids, the store id and the item id, is this possible in isar? Or does isar support embedded ids? Another approach was search where id==itemid and storeid==storeid, but that takes a lot of time.

Also, I tried putting it in a loop one by one instead of using putAll() and that is taking a tremendous amount of time (2,000,000) records are updated per day (approx 80-90k per hour). Would isolates help here?
